From d85be0999f37ff20ac6b13b1325271ed68556d2e Mon Sep 17 00:00:00 2001
From: Simone Schwizer <simone.schwizer@bytesatwork.ch>
Date: Thu, 13 Mar 2025 15:13:17 +0100
Subject: [PATCH] fdts: stm32mp2: Introduce bytedevkit board
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add the device tree file to support the bytesatwork's byteDEVKIT
OSM.

Signed-off-by: Stephan Dünner <stephan.duenner@bytesatwork.ch>
Signed-off-by: Simone Schwizer <simone.schwizer@bytesatwork.ch>
---
 ...mp25-lpddr4-1x8Gbits-1x16bits-1200MHz.dtsi | 243 ++++++++++++++++++
 fdts/stm32mp257f-bytedevkit-fw-config.dts     |  11 +
 fdts/stm32mp257f-bytedevkit.dts               |  57 ++++
 fdts/stm32mp257f-byteengine.dtsi              | 214 +++++++++++++++
 plat/st/stm32mp2/stm32mp2_def.h               |   8 +-
 5 files changed, 529 insertions(+), 4 deletions(-)
 create mode 100644 fdts/stm32mp25-lpddr4-1x8Gbits-1x16bits-1200MHz.dtsi
 create mode 100644 fdts/stm32mp257f-bytedevkit-fw-config.dts
 create mode 100644 fdts/stm32mp257f-bytedevkit.dts
 create mode 100644 fdts/stm32mp257f-byteengine.dtsi

diff --git a/fdts/stm32mp25-lpddr4-1x8Gbits-1x16bits-1200MHz.dtsi b/fdts/stm32mp25-lpddr4-1x8Gbits-1x16bits-1200MHz.dtsi
new file mode 100644
index 000000000..69446dce0
--- /dev/null
+++ b/fdts/stm32mp25-lpddr4-1x8Gbits-1x16bits-1200MHz.dtsi
@@ -0,0 +1,243 @@
+/*
+ * Copyright (C) 2015-2018, STMicroelectronics - All Rights Reserved
+ *
+ * SPDX-License-Identifier:	GPL-2.0-or-later	BSD-3-Clause
+ *
+ */
+
+/*
+ * File generated by STMicroelectronics STM32CubeMX DDR Tool for MPUs
+ * DDR type: LPDDR4
+ * DDR width: 16bits
+ * DDR density: 8Gb
+ * System frequency: 1200000kHz
+ * Address mapping type: RBC
+ *
+ * Save Date: 2025.01.22, save Time: 11:26:59
+ */
+
+ #define DDR_MEM_NAME "LPDDR4 16bits 1200000kHz"
+ #define DDR_MEM_SPEED 1200000
+ #define DDR_MEM_SIZE 0x40000000
+ 
+ #define DDR_MSTR 0x01081020
+ #define DDR_MRCTRL0 0x00000030
+ #define DDR_MRCTRL1 0x00000000
+ #define DDR_MRCTRL2 0x00000000
+ #define DDR_DERATEEN 0x00000203
+ #define DDR_DERATEINT 0x0124f800
+ #define DDR_DERATECTL 0x00000000
+ #define DDR_PWRCTL 0x00000100
+ #define DDR_PWRTMG 0x00130001
+ #define DDR_HWLPCTL 0x00000002
+ #define DDR_RFSHCTL0 0x00210014
+ #define DDR_RFSHCTL1 0x00000000
+ #define DDR_RFSHCTL3 0x00000000
+ #define DDR_RFSHTMG 0x81240054
+ #define DDR_RFSHTMG1 0x00360000
+ #define DDR_CRCPARCTL0 0x00000000
+ #define DDR_CRCPARCTL1 0x00001000
+ #define DDR_INIT0 0xc0020002
+ #define DDR_INIT1 0x00010002
+ #define DDR_INIT2 0x00000d00
+ #define DDR_INIT3 0x00c40024
+ #define DDR_INIT4 0x00310008
+ #define DDR_INIT5 0x00100004
+ #define DDR_INIT6 0x00660047
+ #define DDR_INIT7 0x00050047
+ #define DDR_DIMMCTL 0x00000000
+ #define DDR_RANKCTL 0x0000066f
+ #define DDR_RANKCTL1 0x00000011
+ #define DDR_DRAMTMG0 0x1718141a
+ #define DDR_DRAMTMG1 0x00050524
+ #define DDR_DRAMTMG2 0x060c1111
+ #define DDR_DRAMTMG3 0x0090900c
+ #define DDR_DRAMTMG4 0x0b04060b
+ #define DDR_DRAMTMG5 0x02030909
+ #define DDR_DRAMTMG6 0x02020007
+ #define DDR_DRAMTMG7 0x00000302
+ #define DDR_DRAMTMG8 0x03034405
+ #define DDR_DRAMTMG9 0x0004040d
+ #define DDR_DRAMTMG10 0x001c180a
+ #define DDR_DRAMTMG11 0x440c021c
+ #define DDR_DRAMTMG12 0x1a020010
+ #define DDR_DRAMTMG13 0x0b100002
+ #define DDR_DRAMTMG14 0x000000ad
+ #define DDR_DRAMTMG15 0x00000000
+ #define DDR_ZQCTL0 0x02580012
+ #define DDR_ZQCTL1 0x01e0493e
+ #define DDR_ZQCTL2 0x00000000
+ #define DDR_DFITMG0 0x0395820a
+ #define DDR_DFITMG1 0x000a0303
+ #define DDR_DFILPCFG0 0x07f04111
+ #define DDR_DFILPCFG1 0x000000f0
+ #define DDR_DFIUPD0 0x4040000c
+ #define DDR_DFIUPD1 0x0040007f
+ #define DDR_DFIUPD2 0x00000000
+ #define DDR_DFIMISC 0x00000041
+ #define DDR_DFITMG2 0x0000150a
+ #define DDR_DFITMG3 0x00000000
+ #define DDR_DBICTL 0x00000001
+ #define DDR_DFIPHYMSTR 0x80000001
+ #define DDR_ADDRMAP0 0x0000001f
+ #define DDR_ADDRMAP1 0x00070707
+ #define DDR_ADDRMAP2 0x00000000
+ #define DDR_ADDRMAP3 0x1f000000
+ #define DDR_ADDRMAP4 0x00001f1f
+ #define DDR_ADDRMAP5 0x060f0606
+ #define DDR_ADDRMAP6 0x06060606
+ #define DDR_ADDRMAP7 0x00000f0f
+ #define DDR_ADDRMAP8 0x00003f3f
+ #define DDR_ADDRMAP9 0x06060606
+ #define DDR_ADDRMAP10 0x06060606
+ #define DDR_ADDRMAP11 0x00000006
+ #define DDR_ODTCFG 0x04000400
+ #define DDR_ODTMAP 0x00000000
+ #define DDR_SCHED 0x80001b00
+ #define DDR_SCHED1 0x00000000
+ #define DDR_PERFHPR1 0x04000200
+ #define DDR_PERFLPR1 0x08000080
+ #define DDR_PERFWR1 0x08000400
+ #define DDR_SCHED3 0x04040208
+ #define DDR_SCHED4 0x08400810
+ #define DDR_DBG0 0x00000000
+ #define DDR_DBG1 0x00000000
+ #define DDR_DBGCMD 0x00000000
+ #define DDR_SWCTL 0x00000000
+ #define DDR_SWCTLSTATIC 0x00000000
+ #define DDR_POISONCFG 0x00000000
+ #define DDR_PCCFG 0x00000000
+ #define DDR_PCFGR_0 0x00704100
+ #define DDR_PCFGW_0 0x00004100
+ #define DDR_PCTRL_0 0x00000000
+ #define DDR_PCFGQOS0_0 0x0021000c
+ #define DDR_PCFGQOS1_0 0x01000080
+ #define DDR_PCFGWQOS0_0 0x01100c07
+ #define DDR_PCFGWQOS1_0 0x04000200
+ #define DDR_PCFGR_1 0x00704100
+ #define DDR_PCFGW_1 0x00004100
+ #define DDR_PCTRL_1 0x00000000
+ #define DDR_PCFGQOS0_1 0x00100007
+ #define DDR_PCFGQOS1_1 0x01000080
+ #define DDR_PCFGWQOS0_1 0x01100c07
+ #define DDR_PCFGWQOS1_1 0x04000200
+ #define DDR_UIB_DRAMTYPE 0x00000002
+ #define DDR_UIB_DIMMTYPE 0x00000004
+ #define DDR_UIB_LP4XMODE 0x00000000
+ #define DDR_UIB_NUMDBYTE 0x00000004
+ #define DDR_UIB_NUMACTIVEDBYTEDFI0 0x00000002
+ #define DDR_UIB_NUMACTIVEDBYTEDFI1 0x00000000
+ #define DDR_UIB_NUMANIB 0x00000008
+ #define DDR_UIB_NUMRANK_DFI0 0x00000001
+ #define DDR_UIB_NUMRANK_DFI1 0x00000001
+ #define DDR_UIB_DRAMDATAWIDTH 0x00000010
+ #define DDR_UIB_NUMPSTATES 0x00000001
+ #define DDR_UIB_FREQUENCY_0 0x000004b0
+ #define DDR_UIB_PLLBYPASS_0 0x00000000
+ #define DDR_UIB_DFIFREQRATIO_0 0x00000001
+ #define DDR_UIB_DFI1EXISTS 0x00000001
+ #define DDR_UIB_TRAIN2D 0x00000000
+ #define DDR_UIB_HARDMACROVER 0x00000003
+ #define DDR_UIB_READDBIENABLE_0 0x00000000
+ #define DDR_UIB_DFIMODE 0x00000000
+ #define DDR_UIA_LP4RXPREAMBLEMODE_0 0x00000000
+ #define DDR_UIA_LP4POSTAMBLEEXT_0 0x00000001
+ #define DDR_UIA_D4RXPREAMBLELENGTH_0 0x00000001
+ #define DDR_UIA_D4TXPREAMBLELENGTH_0 0x00000000
+ #define DDR_UIA_EXTCALRESVAL 0x00000000
+ #define DDR_UIA_IS2TTIMING_0 0x00000000
+ #define DDR_UIA_ODTIMPEDANCE_0 0x00000035
+ #define DDR_UIA_TXIMPEDANCE_0 0x00000028
+ #define DDR_UIA_ATXIMPEDANCE 0x00000028
+ #define DDR_UIA_MEMALERTEN 0x00000000
+ #define DDR_UIA_MEMALERTPUIMP 0x00000000
+ #define DDR_UIA_MEMALERTVREFLEVEL 0x00000000
+ #define DDR_UIA_MEMALERTSYNCBYPASS 0x00000000
+ #define DDR_UIA_DISDYNADRTRI_0 0x00000001
+ #define DDR_UIA_PHYMSTRTRAININTERVAL_0 0x0000000a
+ #define DDR_UIA_PHYMSTRMAXREQTOACK_0 0x00000005
+ #define DDR_UIA_WDQSEXT 0x00000001
+ #define DDR_UIA_CALINTERVAL 0x00000009
+ #define DDR_UIA_CALONCE 0x00000000
+ #define DDR_UIA_LP4RL_0 0x00000004
+ #define DDR_UIA_LP4WL_0 0x00000004
+ #define DDR_UIA_LP4WLS_0 0x00000000
+ #define DDR_UIA_LP4DBIRD_0 0x00000000
+ #define DDR_UIA_LP4DBIWR_0 0x00000000
+ #define DDR_UIA_LP4NWR_0 0x00000004
+ #define DDR_UIA_LP4LOWPOWERDRV 0x00000000
+ #define DDR_UIA_DRAMBYTESWAP 0x00000000
+ #define DDR_UIA_RXENBACKOFF 0x00000000
+ #define DDR_UIA_TRAINSEQUENCECTRL 0x00000000
+ #define DDR_UIA_SNPSUMCTLOPT 0x00000000
+ #define DDR_UIA_SNPSUMCTLF0RC5X_0 0x00000000
+ #define DDR_UIA_TXSLEWRISEDQ_0 0x0000000f
+ #define DDR_UIA_TXSLEWFALLDQ_0 0x0000000f
+ #define DDR_UIA_TXSLEWRISEAC 0x0000000f
+ #define DDR_UIA_TXSLEWFALLAC 0x0000000f
+ #define DDR_UIA_DISABLERETRAINING 0x00000000
+ #define DDR_UIA_DISABLEPHYUPDATE 0x00000001
+ #define DDR_UIA_ENABLEHIGHCLKSKEWFIX 0x00000000
+ #define DDR_UIA_DISABLEUNUSEDADDRLNS 0x00000001
+ #define DDR_UIA_PHYINITSEQUENCENUM 0x00000000
+ #define DDR_UIA_ENABLEDFICSPOLARITYFIX 0x00000000
+ #define DDR_UIA_PHYVREF 0x00000014
+ #define DDR_UIA_SEQUENCECTRL_0 0x0000131f
+ #define DDR_UIM_MR0_0 0x00000000
+ #define DDR_UIM_MR1_0 0x000000c4
+ #define DDR_UIM_MR2_0 0x00000024
+ #define DDR_UIM_MR3_0 0x00000031
+ #define DDR_UIM_MR4_0 0x00000000
+ #define DDR_UIM_MR5_0 0x00000000
+ #define DDR_UIM_MR6_0 0x00000000
+ #define DDR_UIM_MR11_0 0x00000066
+ #define DDR_UIM_MR12_0 0x00000047
+ #define DDR_UIM_MR13_0 0x00000008
+ #define DDR_UIM_MR14_0 0x00000047
+ #define DDR_UIM_MR22_0 0x00000005
+ #define DDR_UIS_SWIZZLE_0 0x00000003
+ #define DDR_UIS_SWIZZLE_1 0x00000002
+ #define DDR_UIS_SWIZZLE_2 0x00000000
+ #define DDR_UIS_SWIZZLE_3 0x00000001
+ #define DDR_UIS_SWIZZLE_4 0x00000006
+ #define DDR_UIS_SWIZZLE_5 0x00000007
+ #define DDR_UIS_SWIZZLE_6 0x00000005
+ #define DDR_UIS_SWIZZLE_7 0x00000004
+ #define DDR_UIS_SWIZZLE_8 0x00000005
+ #define DDR_UIS_SWIZZLE_9 0x00000004
+ #define DDR_UIS_SWIZZLE_10 0x00000007
+ #define DDR_UIS_SWIZZLE_11 0x00000006
+ #define DDR_UIS_SWIZZLE_12 0x00000000
+ #define DDR_UIS_SWIZZLE_13 0x00000003
+ #define DDR_UIS_SWIZZLE_14 0x00000002
+ #define DDR_UIS_SWIZZLE_15 0x00000001
+ #define DDR_UIS_SWIZZLE_16 0x00000005
+ #define DDR_UIS_SWIZZLE_17 0x00000007
+ #define DDR_UIS_SWIZZLE_18 0x00000006
+ #define DDR_UIS_SWIZZLE_19 0x00000004
+ #define DDR_UIS_SWIZZLE_20 0x00000000
+ #define DDR_UIS_SWIZZLE_21 0x00000001
+ #define DDR_UIS_SWIZZLE_22 0x00000003
+ #define DDR_UIS_SWIZZLE_23 0x00000002
+ #define DDR_UIS_SWIZZLE_24 0x00000007
+ #define DDR_UIS_SWIZZLE_25 0x00000004
+ #define DDR_UIS_SWIZZLE_26 0x00000005
+ #define DDR_UIS_SWIZZLE_27 0x00000006
+ #define DDR_UIS_SWIZZLE_28 0x00000002
+ #define DDR_UIS_SWIZZLE_29 0x00000003
+ #define DDR_UIS_SWIZZLE_30 0x00000001
+ #define DDR_UIS_SWIZZLE_31 0x00000000
+ #define DDR_UIS_SWIZZLE_32 0x00000000
+ #define DDR_UIS_SWIZZLE_33 0x00000001
+ #define DDR_UIS_SWIZZLE_34 0x00000002
+ #define DDR_UIS_SWIZZLE_35 0x00000003
+ #define DDR_UIS_SWIZZLE_36 0x00000004
+ #define DDR_UIS_SWIZZLE_37 0x00000005
+ #define DDR_UIS_SWIZZLE_38 0x00000000
+ #define DDR_UIS_SWIZZLE_39 0x00000001
+ #define DDR_UIS_SWIZZLE_40 0x00000002
+ #define DDR_UIS_SWIZZLE_41 0x00000003
+ #define DDR_UIS_SWIZZLE_42 0x00000004
+ #define DDR_UIS_SWIZZLE_43 0x00000005
+
+ #include "stm32mp25-ddr.dtsi"
\ No newline at end of file
diff --git a/fdts/stm32mp257f-bytedevkit-fw-config.dts b/fdts/stm32mp257f-bytedevkit-fw-config.dts
new file mode 100644
index 000000000..0c5b3729f
--- /dev/null
+++ b/fdts/stm32mp257f-bytedevkit-fw-config.dts
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2025 bytesatwork AG - https://www.bytesatwork.io
+ *
+ * Based on stm32mp257f-dk-fw-config.dts
+ *
+ * Copyright (c) 2020-2023, STMicroelectronics - All Rights Reserved
+ */
+
+#include "stm32mp25-fw-config.dtsi"
+#include "stm32mp257f-dk-ca35tdcid-fw-config.dtsi"
diff --git a/fdts/stm32mp257f-bytedevkit.dts b/fdts/stm32mp257f-bytedevkit.dts
new file mode 100644
index 000000000..672847f80
--- /dev/null
+++ b/fdts/stm32mp257f-bytedevkit.dts
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2025 bytesatwork AG - https://www.bytesatwork.io
+ *
+ * Based on stm32mp257f-dk.dts
+ *
+ * Copyright (C) STMicroelectronics 2021 - All Rights Reserved
+ */
+
+/dts-v1/;
+
+#include "stm32mp257f-byteengine.dtsi"
+
+/ {
+	model = "byteDEVKIT";
+	compatible = "st,stm32mp257f-bytedevkit", "st,stm32mp257";
+
+	aliases {
+		serial0 = &usart1;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+};
+
+&pinctrl {
+	/omit-if-no-ref/
+	usart1_pins: usart1-0 {
+			pins1 {
+					pinmux = <STM32_PINMUX('G', 14, AF6)>; /* USART1_TX */
+					bias-disable;
+					drive-push-pull;
+					slew-rate = <0>;
+			};
+			pins2 {
+					pinmux = <STM32_PINMUX('G', 15, AF6)>; /* USART1_RX */
+					bias-disable;
+			};
+	};
+};
+
+&sdmmc1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&sdmmc1_b4_pins_b>;
+	st,neg-edge;
+	bus-width = <4>;
+	vmmc-supply = <&vdd_sdcard>;
+	vqmmc-supply = <&vddio1>;
+	status = "okay";
+};
+
+&usart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&usart1_pins>;
+	status = "okay";
+};
diff --git a/fdts/stm32mp257f-byteengine.dtsi b/fdts/stm32mp257f-byteengine.dtsi
new file mode 100644
index 000000000..f85419e0c
--- /dev/null
+++ b/fdts/stm32mp257f-byteengine.dtsi
@@ -0,0 +1,214 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2025 bytesatwork AG - https://www.bytesatwork.io
+ *
+ * Based on stm32mp257f-dk.dts
+ *
+ * Copyright (C) STMicroelectronics 2021 - All Rights Reserved
+ */
+
+#include <dt-bindings/clock/stm32mp25-clksrc.h>
+#include "stm32mp257.dtsi"
+#include "stm32mp25xf.dtsi"
+#include "stm32mp257f-dk-ca35tdcid-rcc.dtsi"
+#include "stm32mp25-lpddr4-1x8Gbits-1x16bits-1200MHz.dtsi"
+#include "stm32mp25-pinctrl.dtsi"
+#include "stm32mp25xxal-pinctrl.dtsi"
+
+/ {
+	memory@80000000 {
+		device_type = "memory";
+		reg = <0x0 0x80000000 0x0 0x40000000>;
+	};
+
+	shadow-prov {
+		compatible = "st,provisioning";
+
+		hconf1_prov {
+			nvmem-cells = <&hconf1_otp>;
+			st,shadow-value = <0x00018000>;
+		};
+	};
+};
+
+&pinctrl {
+	/omit-if-no-ref/
+	i2c4_pins: i2c4-0 {
+		pins {
+				pinmux = <STM32_PINMUX('C', 6, AF9)>, /* I2C4_SCL */
+						 <STM32_PINMUX('A', 1, AF8)>; /* I2C4_SDA */
+				bias-pull-up;
+				drive-open-drain;
+				slew-rate = <0>;
+		};
+	};
+};
+
+&bsec {
+	board_id: board_id@3d8 {
+		reg = <0x3d8 0x4>;
+	};
+};
+
+&ddr {
+	vdd1-supply = <&vdd1_ddr>;
+	vdd2-supply = <&vdd2_ddr>;
+	vddq-supply = <&vdd2_ddr>;
+};
+
+&hash {
+	status = "okay";
+};
+
+&i2c4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c4_pins>;
+	i2c-scl-rising-time-ns = <185>;
+	i2c-scl-falling-time-ns = <20>;
+	clock-frequency = <400000>;
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	secure-status = "okay";
+
+	pmic2: stpmic@33 {
+		compatible = "st,stpmic2";
+		reg = <0x33>;
+		status = "okay";
+
+		regulators {
+			compatible = "st,stpmic2-regulators";
+
+			vddcpu: buck1 {
+				regulator-name = "vddcpu";
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <910000>;
+				regulator-always-on;
+			};
+			vddcore: buck2 {
+				regulator-name = "vddcore";
+				regulator-min-microvolt = <820000>;
+				regulator-max-microvolt = <820000>;
+				regulator-always-on;
+			};
+			vddgpu: buck3 {
+				regulator-name = "vddgpu";
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <900000>;
+				regulator-always-on;
+			};
+			vddio_pmic: buck4 {
+				regulator-name = "vddio_pmic";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-always-on;
+			};
+			v1v8: buck5 {
+				regulator-name = "v1v8";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-always-on;
+			};
+			vdd2_ddr: buck6 {
+				regulator-name = "vdd2_ddr";
+				regulator-min-microvolt = <1100000>;
+				regulator-max-microvolt = <1100000>;
+			};
+			v3v3: buck7 {
+				regulator-name = "v3v3";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+			vdda1v8_aon: ldo1 {
+				regulator-name = "vdda1v8_aon";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-always-on;
+			};
+			vdd_emmc: ldo2 {
+				regulator-name = "vdd_emmc";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+			vdd1_ddr: ldo3 {
+				regulator-name = "vdd1_ddr";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-enable-ramp-delay = <1000>;
+			};
+			vdd3v3_usb: ldo4 {
+				regulator-name = "vdd3v3_usb";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+			v5v_hdmi: ldo5 {
+				regulator-name = "v5v_hdmi";
+				regulator-min-microvolt = <2000000>;
+				regulator-max-microvolt = <2000000>;
+			};
+			vdd_sdcard: ldo7 {
+				regulator-name = "vdd_sdcard";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+			vddio_sdcard: ldo8 {
+				regulator-name = "vddio_sdcard";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+		};
+	};
+};
+
+&iwdg1 {
+	timeout-sec = <32>;
+	status = "okay";
+};
+
+&pwr {
+	vddio1: vddio1 {
+		vddio1-supply = <&vddio_sdcard>;
+	};
+	vddio2: vddio2 {
+		vddio2-supply = <&v1v8>;
+	};
+	vddio3: vddio3 {
+		vddio3-supply = <&vddio_pmic>;
+	};
+	vddio4: vddio4 {
+		vddio4-supply = <&vddio_pmic>;
+	};
+	vddio: vddio {
+		vdd-supply = <&vddio_pmic>;
+	};
+};
+
+&pka {
+	status = "okay";
+};
+
+&rng {
+	status = "okay";
+};
+
+&saes {
+	status = "okay";
+};
+
+&sdmmc2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&sdmmc2_b4_pins_a &sdmmc2_d47_pins_a>;
+	non-removable;
+	no-sd;
+	no-sdio;
+	st,neg-edge;
+	bus-width = <8>;
+	vmmc-supply = <&vdd_emmc>;
+	vqmmc-supply = <&vddio2>;
+	status = "okay";
+};
diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index ce5983f34..6c6cdcae6 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -466,11 +466,11 @@ enum ddr_type {
 /* For UART crash console */
 #define STM32MP_DEBUG_USART_CLK_FRQ		64000000
 /* USART2 on HSI@64MHz, TX on GPIOA4 Alternate 6 */
-#define STM32MP_DEBUG_USART_BASE		USART2_BASE
-#define DEBUG_UART_TX_GPIO_BANK_ADDRESS		GPIOA_BASE
-#define DEBUG_UART_TX_GPIO_BANK_CLK_REG		RCC_GPIOACFGR
+#define STM32MP_DEBUG_USART_BASE		USART1_BASE
+#define DEBUG_UART_TX_GPIO_BANK_ADDRESS		GPIOG_BASE
+#define DEBUG_UART_TX_GPIO_BANK_CLK_REG		RCC_GPIOGCFGR
 #define DEBUG_UART_TX_GPIO_BANK_CLK_EN		RCC_GPIOxCFGR_GPIOxEN
-#define DEBUG_UART_TX_GPIO_PORT			4
+#define DEBUG_UART_TX_GPIO_PORT			14
 #define DEBUG_UART_TX_GPIO_ALTERNATE		6
 #define DEBUG_UART_TX_CLKSRC_REG		RCC_XBAR8CFGR
 #define DEBUG_UART_TX_CLKSRC			XBAR_SRC_HSI
